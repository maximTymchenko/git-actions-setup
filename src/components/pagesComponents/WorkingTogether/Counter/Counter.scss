@import '../../../../styles/vars';

.counter {
  width: 100%;
  max-width: 790px;
  margin: 0 auto 80px;
  border-bottom: 2px solid $secondary;

  @media screen and (min-width: $mac__13) {
    position: relative;
    right: 150px;
  }

  @media screen and (max-width: $tablet) {
    padding: 0 10px;
    margin-top: 70px;
  }

  @media screen and (max-width: $mobile) {
    display: none;
  }

  &__header {
    text-align: center;
    display: flex;
    align-items: center;
    justify-content: center;
  }

  &__wrapper {
    display: flex;
    align-items: flex-end;
  }

  &__body {
    box-sizing: border-box;
    position: relative;
    margin-right: 40px;
    margin-top: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 40px;
    background-color: #fff;
    border: 2px solid $secondary;
    max-width: 275px;
    width: 100%;
    height: 230px;

    &::before {
      z-index: -1;
      content: '';
      display: block;
      position: absolute;
      top: 10px;
      left: -15px;
      right: 0;
      bottom: 0;
      height: 200px;
      width: 300px;
      background-color: $primary__light--15;
    }

    &::after {
      z-index: -2;
      content: '';
      display: block;
      position: absolute;
      top: 0px;
      left: -25px;
      right: 0;
      bottom: 0px;
      height: 220px;
      width: 320px;
      background-color: $primary__light--15;
    }
  }

  &__number {
    display: block;
    font-family: 'OktaNeue-Black';
    font-size: 98px;
    font-weight: 900;
    line-height: 0.83;
    color: $secondary;

    &:first-child {
      padding-right: 15px;
    }

    &:last-child {
      padding-left: 15px;
    }
  }

  &__points {
    width: 100%;
    max-width: 790px;
    margin: 60px auto 0;
    display: flex;
    justify-content: space-around;
    align-items: center;
    border-bottom: 2px solid $section__border;
    padding-bottom: 10px;
  }

  &__title {
    font-family: 'OktaNeue-SemiBold';
    font-size: 16px;
    font-weight: 600;
    text-align: center;
    color: $text;
    max-width: 250px;
    width: 100%;
  }

  &__icon {
    cursor: pointer;
    font-size: 22px;
    position: relative;

    &--active {
      fill: $secondary__opacity--50;
    }

    &--rotated {
      cursor: pointer;
      font-size: 22px;
      -webkit-transform: rotateY(180deg);
      transform: rotateY(180deg);
    }
  }
}
