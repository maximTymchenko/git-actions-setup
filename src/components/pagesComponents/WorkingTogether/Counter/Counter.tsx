import React, { FC, useState, useEffect } from 'react';
import SectionTitle from '../../../common/SectionTitle';
import { FiThumbsUp } from 'react-icons/fi';
import './Counter.scss';
import classNames from 'classnames';

interface ICounter {
    leftCounter: number;
    rightCounter: number;
    increaseLeftCount: any;
    increaseRightCount: any;
}
interface ICounterPoints {
    activePoint: null | any;
    points: [
        {
            id: number;
            title: string;
            active: boolean;
        },
    ];
}

const Counter: FC = (props: ICounter) => {
    // const [counterPoints, setCounterPoints] = useState(props.counterPoints);

    const [iconType, setIconType] = useState<boolean>(false);
    const [iconType2, setIconType2] = useState<boolean>(false);

    // //! render counter title and add active class with click
    const [counterPoints, setCounterPoints] = useState<ICounterPoints>({
        activePoint: null,
        points: [
            {
                id: 0,
                title: 'Outcome responsibility',
                active: false,
            },
            {
                id: 1,
                title: 'Project kickoff',
                active: false,
            },
            {
                id: 2,
                title: 'Onboarding approach',
                active: false,
            },
            {
                id: 3,
                title: 'Recruitment & screening',
                active: false,
            },
            {
                id: 4,
                title: 'Remote management',
                active: false,
            },
            {
                id: 5,
                title: 'Communication standards',
                active: false,
            },
            {
                id: 6,
                title: 'Intellectual property',
                active: false,
            },
            {
                id: 7,
                title: 'Pricing & predictability',
                active: false,
            },
        ],
    });

    const togglePointsClass = (index: number) => {
        setCounterPoints({ ...counterPoints, activePoint: counterPoints.points[index] });
    };

    const toggleActivePoints = (index: number) => {
        if (counterPoints.points[index] === counterPoints.activePoint) {
            return 'counter__icon counter__icon--active';
        } else {
            return 'counter__icon';
        }
    };

    const toggleActiveRotatedPoints = (index: number) => {
        if (counterPoints.points[index] === counterPoints.activePoint) {
            return 'counter__icon--rotated counter__icon--active';
        } else {
            return 'counter__icon--rotated';
        }
    };

    return (
        <div data-testid="counter" className="counter">
            <div className="counter__header">
                <SectionTitle type="lighter" className="title" title="Your votes" />
            </div>
            <div className="counter__wrapper">
                <SectionTitle type="secondary" title="On-Demand Team Extension" />
                <div className="counter__body">
                    <span className="counter__number">{props.leftCounter}</span>
                    <span className="counter__number">:</span>
                    <span className="counter__number">{props.rightCounter}</span>
                </div>
                <SectionTitle type="secondary" title="Fully Managed It Services" />
            </div>
            {counterPoints.points.map((point, index) => (
                <div key={index} className="counter__points">
                    <div onClick={props.increaseLeftCount}>
                        <FiThumbsUp
                            // onClick={() => {
                            //     togglePointsClass(index);
                            //     updateStateByIndex();
                            // }}
                            onClick={point.func}
                            // className={toggleActiveRotatedPoints(index)}
                            className={classNames('counter__icon--rotated', {
                                'counter__icon--active': iconType,
                            })}
                        />
                    </div>
                    <span className="counter__title">{point.title}</span>
                    <div onClick={props.increaseRightCount}>
                        <FiThumbsUp
                            // onClick={() => {
                            //     togglePointsClass(index);
                            //     // updateStateByIndex();
                            // }}
                            onClick={point.func}
                            className={classNames('counter__icon--rotated', {
                                'counter__icon--active': iconType,
                            })}
                            // className={toggleActivePoints(index)}
                        />
                    </div>
                </div>
            ))}
        </div>
    );
};
export default Counter;
