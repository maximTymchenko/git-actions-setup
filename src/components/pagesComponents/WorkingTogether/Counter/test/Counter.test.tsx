import React from 'react';
import Counter from '../Counter';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

let counterPoints = [
    {
        id: 0,
        title: 'Outcome responsibility',
        isActive: false,
    },
    {
        id: 1,
        title: 'Project kickoff',
        isActive: false,
    },
    {
        id: 2,
        title: 'Onboarding approach',
        isActive: false,
    },
    {
        id: 3,
        title: 'Recruitment & screening',
        isActive: false,
    },
    {
        id: 4,
        title: 'Remote management',
        isActive: false,
    },
    {
        id: 5,
        title: 'Communication standards',
        isActive: false,
    },
    {
        id: 6,
        title: 'Intellectual property',
        isActive: false,
    },
    {
        id: 7,
        title: 'Pricing & predictability',
        isActive: false,
    },
];

it('renders elem correctly', () => {
    const { getByTestId } = render(<Counter counterPoints={counterPoints} />);
    expect(getByTestId('counter')).toBeTruthy();
});

it('renders without unexpected changes', () => {
    const tree = renderer.create(<Counter counterPoints={counterPoints} />).toJSON();
    expect(tree).toMatchSnapshot();
});
