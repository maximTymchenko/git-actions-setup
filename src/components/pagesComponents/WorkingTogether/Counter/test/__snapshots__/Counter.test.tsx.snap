// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without unexpected changes 1`] = `
<div
  className="counter"
  data-testid="counter"
>
  <div
    className="counter__header"
  >
    <h2
      className="title title--lighter"
      data-testid="section-title"
    >
      Your votes
    </h2>
  </div>
  <div
    className="counter__wrapper"
  >
    <h2
      className="title title--secondary"
      data-testid="section-title"
    >
      On-Demand Team Extension
    </h2>
    <div
      className="counter__body"
    >
      <span
        className="counter__number"
      />
      <span
        className="counter__number"
      >
        :
      </span>
      <span
        className="counter__number"
      />
    </div>
    <h2
      className="title title--secondary"
      data-testid="section-title"
    >
      Fully Managed It Services
    </h2>
  </div>
  <div
    className="counter__points"
  >
    <div>
      <svg
        className="counter__icon--rotated"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"
        />
      </svg>
    </div>
    <span
      className="counter__title"
    >
      Outcome responsibility
    </span>
    <div>
      <svg
        className="counter__icon--rotated"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"
        />
      </svg>
    </div>
  </div>
  <div
    className="counter__points"
  >
    <div>
      <svg
        className="counter__icon--rotated"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"
        />
      </svg>
    </div>
    <span
      className="counter__title"
    >
      Project kickoff
    </span>
    <div>
      <svg
        className="counter__icon--rotated"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"
        />
      </svg>
    </div>
  </div>
  <div
    className="counter__points"
  >
    <div>
      <svg
        className="counter__icon--rotated"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"
        />
      </svg>
    </div>
    <span
      className="counter__title"
    >
      Onboarding approach
    </span>
    <div>
      <svg
        className="counter__icon--rotated"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"
        />
      </svg>
    </div>
  </div>
  <div
    className="counter__points"
  >
    <div>
      <svg
        className="counter__icon--rotated"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"
        />
      </svg>
    </div>
    <span
      className="counter__title"
    >
      Recruitment & screening
    </span>
    <div>
      <svg
        className="counter__icon--rotated"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"
        />
      </svg>
    </div>
  </div>
  <div
    className="counter__points"
  >
    <div>
      <svg
        className="counter__icon--rotated"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"
        />
      </svg>
    </div>
    <span
      className="counter__title"
    >
      Remote management
    </span>
    <div>
      <svg
        className="counter__icon--rotated"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"
        />
      </svg>
    </div>
  </div>
  <div
    className="counter__points"
  >
    <div>
      <svg
        className="counter__icon--rotated"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"
        />
      </svg>
    </div>
    <span
      className="counter__title"
    >
      Communication standards
    </span>
    <div>
      <svg
        className="counter__icon--rotated"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"
        />
      </svg>
    </div>
  </div>
  <div
    className="counter__points"
  >
    <div>
      <svg
        className="counter__icon--rotated"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"
        />
      </svg>
    </div>
    <span
      className="counter__title"
    >
      Intellectual property
    </span>
    <div>
      <svg
        className="counter__icon--rotated"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"
        />
      </svg>
    </div>
  </div>
  <div
    className="counter__points"
  >
    <div>
      <svg
        className="counter__icon--rotated"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"
        />
      </svg>
    </div>
    <span
      className="counter__title"
    >
      Pricing & predictability
    </span>
    <div>
      <svg
        className="counter__icon--rotated"
        fill="none"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14 9V5a3 3 0 0 0-3-3l-4 9v11h11.28a2 2 0 0 0 2-1.7l1.38-9a2 2 0 0 0-2-2.3zM7 22H4a2 2 0 0 1-2-2v-7a2 2 0 0 1 2-2h3"
        />
      </svg>
    </div>
  </div>
</div>
`;
